src/scales/scale.category.js

```diff
@@ -53,10 +53,15 @@ module.exports = function() {
 
 		getLabelForIndex: function(index, datasetIndex) {
 			var me = this;
-			var data = me.chart.data;
+			var chart = me.chart;
+			var data = chart.data;
 			var isHorizontal = me.isHorizontal();
+			var ds = typeof chart.getDatasetMeta === 'function' && chart.getDatasetMeta(datasetIndex).controller;
+			var isValue = typeof ds.getValueScaleId === 'function'
```

Comment:
Could this be written more concisely as:

    var isValue =  ds.getValueScaleId && ds.getValueScaleId() === me.id

Reply:
Almost, changed and added comments.
Also typedef is a bit slower in some brosers: https://jsperf.com/typeof-versus-plain-if